test/models/apple/publisher_test.rb

```diff
@@ -369,6 +405,70 @@
     end
   end
 
+  describe "#increment_asset_wait!" do
+    let(:episode1) { build(:uploaded_apple_episode, show: apple_publisher.show) }
+    let(:episode2) { build(:uploaded_apple_episode, show: apple_publisher.show) }
+    let(:episodes) { [episode1, episode2] }
+
+    it "should increment asset wait count for each episode" do
+      episodes.each do |ep|
+        assert_equal 0, ep.apple_episode_delivery_status.asset_processing_attempts
+      end
+
+      apple_publisher.increment_asset_wait!(episodes)
+
+      episodes.each do |ep|
+        assert_equal 1, ep.apple_episode_delivery_status.asset_processing_attempts
+      end
+    end
+
+    it "should only increment the episodes that are still waiting" do
+      assert 1, episode1.podcast_delivery_files.length
+      assert 1, episode2.podcast_delivery_files.length
+
+      episode2.podcast_delivery_files.first.stub(:api_marked_as_uploaded?, false) do
+        episode1.podcast_delivery_files.first.stub(:api_marked_as_uploaded?, true) do
+          apple_publisher.increment_asset_wait!(episodes)
+        end
+      end
+
+      assert_equal [1, 0], [episode1, episode2].map { |ep| ep.apple_episode_delivery_status.asset_processing_attempts }
+    end
+
+    it "logs a timeout message with correct information" do
+      eps = [
+        OpenStruct.new(
+          podcast_delivery_files: [OpenStruct.new(api_marked_as_uploaded?: true)],
+          apple_episode_delivery_status: OpenStruct.new(
+            increment_asset_wait: nil,
+            asset_processing_attempts: 3
+          )
+        )
+      ] * 2  # Create two identical episode structures
+      eps = eps.map.with_index do |e, i| # Set the feeder_id to a unique value
+        e = e.dup
+        e.feeder_id = i + 1
+        e
+      end
+
+      assert publisher
+      logs = capture_json_logs do
+        Apple::Episode.stub :wait_for_asset_state, [true, eps] do
+          error = assert_raises(RuntimeError) do
+            publisher.wait_for_asset_state(eps)
+          end
+          assert_equal "Timed out waiting for asset state", error.message
+        end
+      end
+
+      log = logs[0]
+      assert_equal "Timed out waiting for asset state", log[:msg]
+      assert_equal 30, log[:level]
```

Comment:
Oh, so this PR is already downgrading these to log.infos?  Where is that happening?

Hrm... not sure I'm seeing it.  Is it a side effect of a change in here?